### PR TITLE
Shared View Page

### PR DESCRIPTION
**Pull request by guptaharsh25**
This PR lets you create shared views of your template. The shared view URL is `{baseURL}/preview/{templateID}/{version}`. For example, the URL can be something like `http://localhost:3000/preview/5e574c4e04a0d8004e0076c3/1.0`.

## Work done

- created a new action in search reducer to toggle visibility of search bar so that it does not show up on shared view page
- reformatted the router definition in App.tsx so that the sidebar does not show up in the shared view page
- created UI for shared view page
- updated AdaptiveCard.tsx so that it can accept an optional prop `version` so that we can specify what version of a template we want to render. if `version` is not specified the latest version of the template is rendered.
- implemented the copy JSON button on Shared View page

## Screenshot

image

## Steps for testing

- Create a new card using the Designer tool in the app
- Save it
- Use postman to find the templateID of the template you created
- Go to URL `http://localhost:3000/preview/{templateID}/1.0
- Press Copy JSON button and paste somewhere to see if it works

## Next steps

- Need to make this page viewable only if this template has been shared by the owner
- Need to make the sharing modal so that owners can share templates

### CODE REVIEW

File: private-templates-service/adaptivecards-templating-client/typescript/api.d.ts [hunk 1]
```diff
@@ -30,6 +31,67 @@ export declare class BaseErrorError {
         type: string;
     }[];
 }
+export declare class PostedTemplate {
```
Comment (guptaharsh25) [inline private-templates-service/adaptivecards-templating-client/typescript/api.d.ts]: Need to investigate why there are large changes in the client SDK on my branch

File: private-templates-service/adaptivecards-templating-client/typescript/api.js [hunk 2]
```diff
@@ -161,6 +161,126 @@ var BaseErrorError = (function () {
     return BaseErrorError;
 }());
 exports.BaseErrorError = BaseErrorError;
+var PostedTemplate = (function () {
```
Comment (guptaharsh25) [inline private-templates-service/adaptivecards-templating-client/typescript/api.js]: Need to investigate why there are large changes in the client SDK on my branch

File: private-templates-service/adaptivecards-templating-client/typescript/api.js.map [hunk 3]
```diff
@@ -1 +1 @@
...
...M,IAAI,KAAK,CAAC,sEAAsE,CAAC,CAAC;SAC3F;QAEK,MAAO,CAAC,MAAM,CAAC,oBAAoB,EAAE,OAAO,CAAC,OAAO,CAAC,CAAC;QAE5D,IAAI,mBAAmB,GAAG,KAAK,CAAC;QAEhC,IAAI,sBAAsB,GAA4B;YAClD,MAAM,EAAE,MAAM;YACd,EAAE,EAAE,uBAAuB;YAC3B,OAAO,EAAE,oBAAoB;YAC7B,GAAG,EAAE,YAAY;YACjB,cAAc,EAAE,IAAI,CAAC,eAAe;YACpC,IAAI,EAAE,IAAI;YACV,IAAI,EAAE,gBAAgB,CAAC,SAAS,CAAC,IAAI,EAAE,YAAY,CAAC;SACvD,CAAC;QAEF,IAAI,CAAC,eAAe,CAAC,WAAW,CAAC,cAAc,CAAC,sBAAsB,CAAC,CAAC;QAExE,IAAI,CAAC,eAAe,CAAC,OAAO,CAAC,cAAc,CAAC,sBAAsB,CAAC,CAAC;QAEpE,IAAI,MAAM,CAAC,IAAI,CAAC,kBAAkB,CAAC,CAAC,MAAM,EAAE;YACxC,IAAI,mBAAmB,EAAE;gBACf,sBAAuB,CAAC,QAAQ,GAAG,kBAAkB,CAAC;aAC/D;iBAAM;gBACH,sBAAsB,CAAC,IAAI,GAAG,kBAAkB,CAAC;aACpD;SACJ;QACD,OAAO,IAAI,OAAO,CAAmD,UAAC,OAAO,EAAE,MAAM;YACjF,eAAe,CAAC,sBAAsB,EAAE,UAAC,KAAK,EAAE,QAAQ,EAAE,IAAI;gBAC1D,IAAI,KAAK,EAAE;oBACP,MAAM,CAAC,KAAK,CAAC,CAAC;iBACjB;qBAAM;oBACH,IAAI,QAAQ,CAAC,UAAU,IAAI,QAAQ,CAAC,UAAU,IAAI,GAAG,IAAI,QAAQ,CAAC,UAAU,IAAI,GAAG,EAAE;wBACjF,OAAO,CAAC,EAAE,QAAQ,EAAE,QAAQ,EAAE,IAAI,EAAE,IAAI,EAAE,CAAC,CAAC;qBAC/C;yBAAM;wBACH,MAAM,CAAC,EAAE,QAAQ,EAAE,QAAQ,EAAE,IAAI,EAAE,IAAI,EAAE,CAAC,CAAC;qBAC9C;iBACJ;YACL,CAAC,CAAC,CAAC;QACP,CAAC,CAAC,CAAC;IACP,CAAC;IACL,cAAC;AAAD,CAAC,AArMD,IAqMC;AArMY,0BAAO"}
```
Comment (guptaharsh25) [inline private-templates-service/adaptivecards-templating-client/typescript/api.js.map]: Need to investigate why there are large changes in the client SDK on my branch

File: private-templates-service/client/src/components/Shared/Shared.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    getTemplate: (templateID: string) => {
+      dispatch(getTemplate(templateID));
+    }
+  };
+};
+
+interface SharedComponentProps {
+  currentPageTitle: string;
+  template: Template;
+  authButtonMethod: () => Promise<void>;
```
Comment (rachel-slater) [inline private-templates-service/client/src/components/Shared/Shared.tsx]: nit: rename to 'authenticate' / 'authButtonClicked'.

File: private-templates-service/client/src/components/Shared/TemplateSourceInfo/TemplateSourceInfo.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+import { TopRowWrapper, TitleWrapper, Title, HeaderWrapper, TemplateSourceWrapper, SourceWrapper, Source } from './styled';
+
+import { ActionButton, ScrollablePane, Text } from 'office-ui-fabric-react';
+
+interface TemplateSourceInfoProps {
+  templateJSON?: string;
+}
+
+const tempJSON = {
```
Comment (rachel-slater) [inline private-templates-service/client/src/components/Shared/TemplateSourceInfo/TemplateSourceInfo.tsx]: ??

File: private-templates-service/client/src/components/Shared/TemplateSourceInfo/TemplateSourceInfo.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  "one": "hello",
+  "two": "bye",
+}
+
+class TemplateSourceInfo extends React.Component<TemplateSourceInfoProps> {
+
+  onCopy = () => {
+    let copyCode = document.createElement('textarea');
+    if (this.props.templateJSON) {
+      let sourceJSON = JSON.stringify(this.props.templateJSON, undefined, 2);
```
Comment (rachel-slater) [inline private-templates-service/client/src/components/Shared/TemplateSourceInfo/TemplateSourceInfo.tsx]: null

**Pull request merged**